src/components/AuthStoreSetter.ts

```diff
@@ -0,0 +1,30 @@
+import _ from 'lodash/fp';
+import { useEffect } from 'react';
+import { AuthContextProps, useAuth } from 'react-oidc-context';
+import { loadAuthToken, OidcUser, processUser } from 'src/libs/auth';
+import { useOnMount } from 'src/libs/react-utils';
+import { authStore } from 'src/libs/state';
+
+function AuthStoreSetter(): null {
+  const auth: AuthContextProps = useAuth();
+
+  useOnMount((): void => {
+    authStore.update(_.set(['authContext'], auth));
+  });
+  useEffect((): void => {
+    const cleanupFns = [
+      auth.events.addUserLoaded((user: OidcUser) => processUser(user, true)),
+      auth.events.addUserUnloaded(() => processUser(null, false)),
+      auth.events.addAccessTokenExpired((): void => {
+        loadAuthToken();
+      }),
+    ];
+    _.over(cleanupFns);
```

Comment:
This is the effect's cleanup function. As such, it must be returned from the effect function.

https://react.dev/reference/react/useEffect#parameters

Reply:
I am running into issues trying to convert this to a callbackFn. It does not seem to like that `_.over(cleanupFns)` returns a void[] when it is expecting a `void` or `() => void`.